### DISCUSSION TITLE
Relationship between PyPDF2 and PyPDF

### DISCUSSION THREAD

**Discussion by vors** [py-pdf/pypdf#2198] [General]
Hello!

I just started using PyPDF2 and it's great, thank you for the awesome tool!

I'm a little confused about the naming though. This github name is pypdf. While it *seems* like the PyPi package that I should be using is https://pypi.org/project/PyPDF2/

While if you google for "PyPDF2 github" it will show you this ancient fork instead of this repo https://github.com/talumbau/PyPDF2

Also there is https://pypdf.readthedocs.io/en/stable/ docs and https://pypdf2.readthedocs.io/en/3.0.0/ (references are the first results form the corresponding google queries "pypdf[2] documentation") which apparently lead to 2 different things?

What's the right way of thinking about all this version difference (and which pages released from where and which one I should be using).

**Comment by stefan6419846** [comment]
There is a page which summarizes the history of PyPDF2 and pypdf inside the docs: https://pypdf.readthedocs.io/en/stable/meta/history.html To keep it short: PyPDF2 development stalled, with some small forks like PyPDF4 appearing. pypdf is the continued effort to keep on maintaining the package under a generic name.

On the PyPI page of PyPDF2 you can read:

> NOTE: The PyPDF2 project is going back to its roots. PyPDF2==3.0.X will be the last version of PyPDF2. Development will continue with [pypdf==3.1.0](https://pypi.org/project/pyPdf/).

As for the docs, there is a corresponding issue to make this more clear as well: #2051.

TL;DR: To get the latest code and a maintained package, you should use the pypdf, which is being developed in this repository.

**Comment by vors** [comment]
Awesome thank you for the clarification and references.

I found that some annotation features are working better in PyPDF2, I will report these as separate issues.

**Comment by vors** [comment]
Added here https://github.com/py-pdf/pypdf/issues/2200
